modules/xphoto/src/inpainting.cpp

```diff
@@ -66,6 +77,820 @@ namespace cv
+                uchar B = bData[y*channels_bgr[0].step1() + x];
+                yData[y*Y.step1() + x] = cv::saturate_cast<uchar>(0.256788235294118 * R + 0.504129411764706 * G + 0.0979058823529412 * B + 16);
+                cbData[y*Cb.step1() + x] = cv::saturate_cast<uchar>(-0.148223529411765 * R - 0.290992156862745* G + 0.439215686274510 * B + 128.0);
+                crData[y*Cr.step1() + x] = cv::saturate_cast<uchar>(0.439215686274510 * R - 0.367788235294118 * G - 0.0714274509803922 * B + 128);
+            }
+        }
+    }
+
+
+    static void
+    icvYCbCr2BGR(cv::Mat& Y, cv::Mat& Cb, cv::Mat& Cr, cv::Mat& bgr)
+    {
+        // same behavior as matlab ycbcr2rgb, when ycbcr image is of type uint8
+        int height = Y.rows;
+        int width = Y.cols;
+
+        uchar* bgrData = (uchar*)bgr.data;
+        double* yData = (double*)Y.data;
+        double* cbData = (double*)Cb.data;
+        double* crData = (double*)Cr.data;
+        for (int y = 0; y < height; ++y)
+        {
+            for (int x = 0; x < width; ++x)
+            {
+                double Y_curr = yData[y*Y.step1() + x];
+                double Cb_curr = cbData[y*Cb.step1() + x];
+                double Cr_curr = crData[y*Cr.step1() + x];
+                bgrData[y*bgr.step1() + 3 * x] = cv::saturate_cast<uchar>(1.16438356164384 * Y_curr + 2.01723263955646* Cb_curr + 0.000003054 * Cr_curr - 276.836305795032);
+                bgrData[y*bgr.step1() + 3 * x + 1] = cv::saturate_cast<uchar>(1.16438356164384 * Y_curr - 0.391762539941450 * Cb_curr - 0.812968292162205 * Cr_curr + 135.575409522967);
+                bgrData[y*bgr.step1() + 3 * x + 2] = cv::saturate_cast<uchar>(1.16438356164384 * Y_curr + 3.01124397411008e-07 * Cb_curr + 1.59602688733570 * Cr_curr - 222.921617109194);
+            }
+        }
+    }
+
+
+    static void
+    icvSgnMat(const cv::Mat& src, cv::Mat& dst) {
+        dst = cv::Mat::zeros(src.size(), CV_64F);
+        double* srcData = (double*)src.data;
+        for (int y = 0; y < src.rows; ++y)
+        {
+            for (int x = 0; x < src.cols; ++x)
+            {
+                double curr_val = srcData[y*src.step1() + x];
+                if (curr_val > 0)
+                {
+                    srcData[y*src.step1() + x] = 1;
+                }
+                else if (curr_val)
+                {
+                    srcData[y*src.step1() + x] = -1;
+                }
+            }
+        }
+    }
+
+
+    static double
+    icvStandardDeviation(const cv::Mat& distorted_block_2d, const cv::Mat& error_mask_2d) {
+        if (cv::countNonZero(error_mask_2d) < 1)
+        {
+            return NAN; // align to matlab behavior
+            // block with no undistorted pixels shouldn't be chosen for processing (only if block_size_min is reached)
+        }
+        cv::Scalar tmp_stddev, tmp_mean;
+        cv::Mat mask8u = error_mask_2d*2;
+        mask8u.convertTo(mask8u, CV_8U);
```

Comment:
> *2

Why 2? What is values range?

> mask8u.convertTo(mask8u, CV_8U);

There is third parameter which can help with multiplication.
BTW, Perhaps inplace mode doesn't work well here.

Reply:
``error_mask_2d`` can have the values 0 (pixels that need to be reconstructed), 0.5 (already reconstructed) or 1 (valid, original pixels). 
By multiplication with 2, the pixels with the value 0.5 won't  be 0 after the conversion to ``CV_8U``. With that, the already reconstructed pixels are used for the computation of the standard deviation. 

Thanks for the remarks.